src/main/scala/firrtl/Utils.scala

```diff
@@ -165,72 +165,54 @@ object Utils extends LazyLogging {
             exps ++ create_exps(WSubIndex(e,i,t.tpe,gender(e))))
       }
    }
+
    def get_flip (t:Type, i:Int, f:Orientation) : Orientation = {
-      if (i >= get_size(t)) error("Shouldn't be here")
-      val x = t match {
-         case (t:UIntType) => f
-         case (t:SIntType) => f
-         case ClockType => f
-         case (t:BundleType) => {
-            var n = i
-            var ret:Option[Orientation] = None
-            t.fields.foreach { x => {
-               if (n < get_size(x.tpe)) {
-                  ret match {
-                     case None => ret = Some(get_flip(x.tpe,n,times(x.flip,f)))
-                     case ret => {}
-                  }
-               } else { n = n - get_size(x.tpe) }
-            }}
-            ret.asInstanceOf[Some[Orientation]].get
-         }
-         case (t:VectorType) => {
-            var n = i
-            var ret:Option[Orientation] = None
-            for (j <- 0 until t.size) {
-               if (n < get_size(t.tpe)) {
-                  ret = Some(get_flip(t.tpe,n,f))
-               } else {
-                  n = n - get_size(t.tpe)
-               }
-            }
-            ret.asInstanceOf[Some[Orientation]].get
+     if (i >= get_size(t)) error("Shouldn't be here")
+     t match {
+       case (_: GroundType) => f
+       case (t: BundleType) => ((t.fields foldLeft (i, None: Option[Orientation])){
+          case ((n, ret), x) if n < get_size(x.tpe) => ret match {
+            case None => (n, Some(get_flip(x.tpe,n,times(x.flip,f))))
+            case Some(_) => (n, ret)
+          }
+          case ((n, ret), x) => (n - get_size(x.tpe), ret)
+       }) match {
+          case (_, Some(p)) => p
+          case (_, None) => error("Shouldn't be here")
+       }
```

Comment:
Rather than matching and checking for some, you could just assign the result of the fold to some value and then check that it is a some via .get or getOrElse error("shouldn't be here") should also probably work.